tests/e2e/editCustodialUserProfileSectionProfile.js

```diff
@@ -0,0 +1,296 @@
+    userProfile.click('@diagnosisType');
+    userProfile.click('@diagnosisTypeGestational');
+    userProfile.click('@save');
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.isSelected('@diagnosisTypeGestational', function (result) {
+      this.assert.equal(typeof result, 'object');
+      this.assert.equal(result.status, 0);
+      this.assert.equal(result.value, true);
+    });
+    userProfile.click('@save');
+  },
+  'Edit and save diagnosis type to prediabetes in custodial user profile page': async (browser) => {
+    const userProfilePage = browser.page.userProfilePage();
+    const userProfile = userProfilePage.section.profile;
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.click('@diagnosisType');
+    userProfile.click('@diagnosisTypePrediabetes');
+    userProfile.click('@save');
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.isSelected('@diagnosisTypePrediabetes', function (result) {
+      this.assert.equal(typeof result, 'object');
+      this.assert.equal(result.status, 0);
+      this.assert.equal(result.value, true);
+    });
+    userProfile.click('@save');
+  },
+  'Edit and save diagnosis type to lada in custodial user profile page': async (browser) => {
+    const userProfilePage = browser.page.userProfilePage();
+    const userProfile = userProfilePage.section.profile;
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.click('@diagnosisType');
+    userProfile.click('@diagnosisTypeLada');
+    userProfile.click('@save');
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.isSelected('@diagnosisTypeLada', function (result) {
+      this.assert.equal(typeof result, 'object');
+      this.assert.equal(result.status, 0);
+      this.assert.equal(result.value, true);
+    });
+    userProfile.click('@save');
+  },
+  'Edit and save diagnosis type to mody in custodial user profile page': async (browser) => {
+    const userProfilePage = browser.page.userProfilePage();
+    const userProfile = userProfilePage.section.profile;
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.click('@diagnosisType');
+    userProfile.click('@diagnosisTypeMody');
+    userProfile.click('@save');
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.isSelected('@diagnosisTypeMody', function (result) {
+      this.assert.equal(typeof result, 'object');
+      this.assert.equal(result.status, 0);
+      this.assert.equal(result.value, true);
+    });
+    userProfile.click('@save');
```

Comment:
```js
    userProfile.isSelected('@diagnosisTypeMody', function (result) {
      this.assert.equal(typeof result, 'object');
      this.assert.equal(result.status, 0);
      this.assert.equal(result.value, true);
    });
```
I don't get it what you are testing here - because it seems that it is same in each (selecting diagnosesType) test. `result.status is 0` and `result.value is true`. 

I don't see other assertion than this one.

Reply:
1. the save at the end is a post-processing step to return the profile page to the original state. The assertion checks after edit/save/edit the diagnosis date has updated (edit is needed for the field to be visible, save hides the field) 
2. .isSelected() determines if an element is selected (the assertion aboved is used in the  documentation: https://nightwatchjs.org/api/isSelected.html )

Reply:
Yeah, I get it why are you doing `post-processing`. I wanted to highlight, that your assertion doesn't work. 
You should check if value is changed after you clicks on `@save` button. Now you are checking only values which you typed into input. But it doesn't mean, that it is safely saved into server and displayed to user.

Flow in automation should be same as user's flow - 
1. Clicking on `Edit` button
2. Edit fullName input
3. Clicking on `Save` button
4. Verifying, that new name is displayed.

What you are doing:
1. Clicking on `Edit` button
2. (Saving value in input `fullName` into variable `previousName`)
3. Edit fullName input 
4. (Saving value in input `fullname` into variable `newName`)
5. Clicking on `Save` button
6. Verifying, that `newName !== `previousName`

Problem is in step 6 - you are verifying values which are in memory. If `Save` button doesn't work, your test will pass due you don't check page after clicking on `Save` button. 


You should do something like that after clicking on `Save`
```js
    browser.assert.equal(await browser.element('.PatientInfo-block--withArrow').getText(), `${previousName}new`);
```